packages/instant-apps-components/src/components/instant-apps-app-guide/instant-apps-app-guide.tsx

```diff
@@ -21,52 +26,72 @@ export class InstantAppsAppGuide {
   @Prop()
   data: AppGuidePage[];
 
+  @Watch('data')
+  watchPropHandler(newValue: AppGuidePage[]) {
+    this.viewModel.pages = newValue;
+    this.el.dispatchEvent(new CustomEvent('calciteCarouselChange'));
+  }
+
   @State() messages: typeof AppGuide_T9n;
+  @State() headerTitle: string;
 
-  viewModel: AppGuideViewModel = new AppGuideViewModel();
+  private viewModel: AppGuideViewModel = new AppGuideViewModel();
+  private carouselRef: HTMLCalciteCarouselElement;
 
-  componentWillRender() {
-    this._initialize();
+  connectedCallback() {
+    this.viewModel.pages = this?.data || [];
```

Comment:
down the road, it might be good to run the data through a sanitizer

Reply:
Added an issue for this
#424